src/openms/source/ANALYSIS/ID/IDRipper.cpp

```diff
@@ -70,34 +73,154 @@ namespace OpenMS
+  IDRipper::IdentificationRuns::IdentificationRuns(const vector<ProteinIdentification>& prot_ids)
+  {
+    for (const auto& prot_id : prot_ids)
+    {
+      String id_run_id = prot_id.getIdentifier();
+      if (this->index_map.find(id_run_id) != this->index_map.end())
+      {
+        throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "IdentificationRun IDs are not unique!", id_run_id);
+      }
+      UInt idx = this->index_map.size();
+      this->index_map[id_run_id] = idx;
+      const DataValue& mv_spectra_data = prot_id.getMetaValue("spectra_data");
+      spectra_data.push_back(mv_spectra_data.isEmpty() ? StringList() : mv_spectra_data.toStringList());
+      }
+  }
+
+  bool IDRipper::RipFileIdentifierIdxComparator::operator()(const RipFileIdentifier& left, const RipFileIdentifier& right) const
+  {
+    return std::tie(left.ident_run_idx, left.file_origin_idx)
+      < std::tie(right.ident_run_idx, right.file_origin_idx);
+  }
+
+  // Identify the output file name features associated via spectra_data or file_origin
+  IDRipper::RipFileIdentifier::RipFileIdentifier(const IDRipper::IdentificationRuns& id_runs, const PeptideIdentification& pep_id, const map<String, UInt>& file_origin_map, const IDRipper::OriginAnnotationFormat origin_annotation_fmt, bool split_ident_runs)
+  {
+      try
+      {
+          // Numerical identifier of the Identification Run
+          this->ident_run_idx   = id_runs.index_map.at(pep_id.getIdentifier());
+
+          // Numerical identifier of the PeptideIdentification origin
+          this->file_origin_idx = (origin_annotation_fmt == MAP_INDEX || origin_annotation_fmt == ID_MERGE_INDEX)
+              ? pep_id.getMetaValue(names_of_OriginAnnotationFormat[origin_annotation_fmt]).toString().toInt()
+              : file_origin_map.at(pep_id.getMetaValue("file_origin").toString());
+
+          // Store the origin full name
+          this->origin_fullname = (origin_annotation_fmt == MAP_INDEX || origin_annotation_fmt == ID_MERGE_INDEX)
+              ? id_runs.spectra_data.at(this->ident_run_idx).at(this->file_origin_idx)
+              : pep_id.getMetaValue("file_origin").toString();
+
+          // Extract the basename, used for output files when --numeric_filenames is not set
+          this->out_basename = QFileInfo(this->origin_fullname.toQString()).completeBaseName().toStdString();
+
+          // Drop the identification run identifier if we're not splitting by identification runs
+          if (!split_ident_runs)
+              this->ident_run_idx = -1u;
+      }
+      catch (const std::out_of_range& e)
+      {
+          throw Exception::ParseError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "input file",
+                  "Failed to identify corresponding spectra_data element for PeptideIdentification element.");
+      }
+  }
+
+  UInt IDRipper::RipFileIdentifier::getIdentRunIdx()
+  {
+      return this->ident_run_idx;
```

Comment:
more a matter of style but I usually don't write this-> in such a context.
only if it could be really ambigious